src/client/transport.js

```diff
@@ -144,77 +141,81 @@ class Transport extends EventEmitter {
         });
     }
 
-    asyncServiceMsg (msg) {
-        return new Promise(resolve => {
-            msg.sessionId = settings.get().sessionId;
+    performRequest (msg, callback) {
+        msg.sessionId = settings.get().sessionId;
 
-            if (isIFrameWithoutSrc)
-                msg.referer = settings.get().referer;
+        if (isIFrameWithoutSrc)
+            msg.referer = settings.get().referer;
 
-            var sendMsg = forced => {
-                this.activeServiceMessagesCounter++;
+        var sendMsg = forced => {
+            this.activeServiceMessagesCounter++;
 
-                var requestIsAsync = this.useAsyncXhr;
+            var requestIsAsync = this.useAsyncXhr;
 
-                if (forced)
-                    requestIsAsync = false;
+            if (forced)
+                requestIsAsync = false;
 
-                var transport    = this;
-                var request      = Transport._createXMLHttpRequest(requestIsAsync);
-                var msgCallback  = function () {
-                    transport.activeServiceMessagesCounter--;
+            var transport    = this;
+            var request      = Transport._createXMLHttpRequest(requestIsAsync);
+            var msgCallback  = function () {
+                transport.activeServiceMessagesCounter--;
 
-                    resolve(this.responseText && parseJSON(this.responseText));
-                };
-                var errorHandler = function () {
-                    if (isWebKit) {
-                        Transport._storeMessage(msg);
-                        msgCallback();
+                callback(this.responseText && parseJSON(this.responseText));
+            };
+            var errorHandler = function () {
+                if (isWebKit) {
+                    Transport._storeMessage(msg);
+                    msgCallback();
+                }
+                else
+                    sendMsg(true);
+            };
+
+            if (forced) {
+                request.addEventListener('readystatechange', function () {
+                    if (this.readyState !== 4)
+                        return;
+
+                    msgCallback();
+                });
+            }
+            else if (isIE9) {
+                //aborted ajax request in IE9 not raise error, abort or timeout events
+                //also getting status code raise error c00c023f
```

Comment:
Aborted ajax requests do not raise the error, abort or timeout events in IE9. Getting a status code raises the c00c023f error.